packages/ui/src/Button/index.tsx

```diff
@@ -0,0 +1,177 @@
+  align-items: center;
+  justify-content: center;
+  color: ${props => props.theme.color.neutral.contrast};
+  cursor: pointer;
+  overflow: hidden;
+  position: relative;
+
+  ${props => (props.$size === 'dense' ? dense : sparse)}
+
+  ${buttonInteractions}
+
+  &::after {
+    outline-offset: -2px;
+  }
+`;
+
+interface BaseButtonProps {
+  /**
+   * The button label. If `iconOnly` is `true`, this will be used as the
+   * `aria-label` attribute.
+   */
+  children: string;
+  /**
+   * Set to `sparse` for more loosely arranged layouts, such as when this is the
+   * submit button for a form. Use `dense` when, e.g., this button appears next
+   * to every item in a dense list of data.
+   *
+   * Default: `sparse`.
+   */
+  size?: Size;
+  /**
+   * What type of action is this button related to? Leave as `default` for most
+   * buttons, set to `accent` for the single most important action on a given
+   * page, set to `unshield` for actions that will unshield the user's funds,
+   * and set to `destructive` for destructive actions.
+   *
+   * Default: `default`
+   */
+  actionType?: ActionType;
+  disabled?: boolean;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+}
+
+interface IconOnlyProps {
+  /**
+   * When `true`, will render just an icon button. The label text passed via
+   * `children` will be used as the `aria-label`.
+   */
+  iconOnly: true;
+  /**
+   * The icon import from `lucide-react` to render. If `iconOnly` is `true`, no
+   * label will be rendered -- just the icon. Otherwise, the icon will be
+   * rendered to the left of the label.
+   *
+   * ```tsx
+   * import { ChevronRight } from 'lucide-react';
+   * <Button icon={ChevronRight}>Label</Button>
+   * <Button icon={ChevronRight} iconOnly>Label</Button>
+   * ```
+   */
+  icon: LucideIcon;
+}
+
+interface RegularProps {
+  /**
+   * When `true`, will render just an icon button. The label text passed via
+   * `children` will be used as the `aria-label`.
+   */
+  iconOnly?: false;
+  /**
+   * The icon import from `lucide-react` to render. If `iconOnly` is `true`, no
+   * label will be rendered -- just the icon. Otherwise, the icon will be
+   * rendered to the left of the label.
+   *
+   * ```tsx
+   * import { ChevronRight } from 'lucide-react';
+   * <Button icon={ChevronRight}>Label</Button>
+   * <Button icon={ChevronRight} iconOnly>Label</Button>
+   * ```
+   */
+  icon?: LucideIcon;
+}
+
+export type ButtonProps = BaseButtonProps & (IconOnlyProps | RegularProps);
+
+/** A component for all your button needs! */
+export const Button = ({
+  children,
+  disabled = false,
+  onClick,
+  icon: IconComponent,
+  iconOnly,
+  size = 'sparse',
+  actionType = 'default',
+}: ButtonProps) => {
+  const priority = useContext(ButtonPriorityContext);
+
+  return (
+    <StyledButton
+      {...asTransientProps({ iconOnly, size, actionType, priority })}
+      disabled={disabled}
+      onClick={onClick}
+      aria-label={iconOnly ? children : undefined}
```

Comment:
suggestion (a11y): do the same as `aria-label` with the `title` attribute, so the text description of the icon button will be visible on long hover

Reply:
Great idea. [Done](https://github.com/penumbra-zone/web/pull/1516/commits/777c06abd4970a4369dd972fbffc8bf07bce9950).